### PR TITLE
feat(nats source): Add log namespace and schema def support

### PR DESCRIPTION
**Pull request by StephenWakely**
Adds log namespace and schema def support.

closes: #15027
epic: #13723

Signed-off-by: Stephen Wakely [fungus.humungus@gmail.com](mailto:fungus.humungus@gmail.com)

### CODE REVIEW

File: src/sources/nats.rs [hunk 1]
```diff
@@ -204,12 +216,64 @@ async fn create_subscription(
...
+        let expected_definition =
+            Definition::new_with_default_metadata(Kind::bytes(), [LogNamespace::Vector])
+                .with_meaning(LookupBuf::root(), "message")
+                .with_metadata_field(&owned_value_path!("vector", "source_type"), Kind::bytes())
+                .with_metadata_field(
+                    &owned_value_path!("vector", "ingest_timestamp"),
+                    Kind::timestamp(),
+                );
+
+        assert_eq!(definition, expected_definition);
```
Comment (spencergilbert) [inline src/sources/nats.rs:251]: I wonder if we should/need to try and test more generically across decoders, or across all decoders - not just the Bytes one. I'm not sure what exactly that would look like without a lot of repetitive typing but throwing it out there.

**Review by spencergilbert — APPROVED** [review]
😅 not a whole lot to this one.

**Review by neuronull — APPROVED** [review]

File: src/sources/nats.rs (hunk 1 above)
Comment (StephenWakely) [reply src/sources/nats.rs:251]: Not sure I understand what you mean? This isn't using Bytes decoder?

Comment (spencergilbert) [reply src/sources/nats.rs:251]: https://vector.dev/docs/reference/configuration/sources/nats/#decoding.codec <- defaults to bytes, therefore this is implicitly using the bytes deserializer. Which is clear to see given the `root()` is the "message", which happens when the deserializer is Bytes.

**Pull request merged**